.github/workflows/ci.yml

```diff
@@ -332,17 +335,127 @@ jobs:
+          echo "PLATFORM_PAIR=${platform//\//-}" >> $GITHUB_ENV
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to the ghcr.io registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Log in to the docker.io registry
+        uses: docker/login-action@v3
+        with:
+          username: fuellabs
+          password: ${{ secrets.DOCKER_IO_READ_ONLY_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY_URL }}
+
+      - name: Setup Rust build cache
+        id: cache
+        uses: buildjet/cache@v3
+        with:
+          path: |
+            home-cargo-bin
+            home-cargo-registry-index
+            home-cargo-registry-cache
+            home-cargo-git-db
+            target
+          key: ${{ env.PLATFORM_PAIR }}-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Inject cache into docker
+        uses: reproducible-containers/buildkit-cache-dance@v3.1.2
+        with:
+          cache-map: |
+            {
+              "home-cargo-bin": "/usr/local/cargo/bin",
+              "home-cargo-registry-index": "/usr/local/cargo/registry/index",
+              "home-cargo-registry-cache": "/usr/local/cargo/registry/cache",
+              "home-cargo-git-db": "/usr/local/cargo/git/db",
+              "target": "/build/target"
+            }
+          skip-extraction: ${{ steps.cache.outputs.cache-hit }}
+
+      - name: Build Docker image
+        id: build
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          platforms: ${{ matrix.arch.platform }}
+          file: deployment/Dockerfile
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=registry,ref=${{ env.REGISTRY_URL }}-build-cache:latest-${{ matrix.arch.runner }}
+          cache-to: type=registry,ref=${{ env.REGISTRY_URL }}-build-cache:latest-${{ matrix.arch.runner }},mode=max,image-manifest=true,oci-mediatypes=true
+          outputs: |
+            type=image,name=${{ env.REGISTRY_URL }},push-by-digest=true,name-canonical=true,push=true
+
+      - name: Export digest
+        run: |
+          mkdir -p /tmp/digests
+          digest="${{ steps.build.outputs.digest }}"
+          touch "/tmp/digests/${digest#sha256:}"
+
+      - name: Upload digest
+        uses: actions/upload-artifact@v4
+        with:
+          name: digests-${{ env.PLATFORM_PAIR }}
+          path: /tmp/digests/*
+          if-no-files-found: error
+          retention-days: 1
+
   publish-docker-image:
     needs:
+      - build-docker-images
       - verifications-complete
-    runs-on: buildjet-4vcpu-ubuntu-2204
+    runs-on: ubuntu-latest
```

Comment:
@Voxelot Do we want to use `buildjet` or switch back to github runners?

Reply:
we should probably stick with buildjet for CI performance reasons

cc @mchristopher

Reply:
FYI, this moves back to basic runners for tasks that are not CPU or memory intensive (like downloading/uploading files). It's an optimization to make sure we're not spending more money than needed on buildjet.